File: mpp/LabMppCurseWPF/LabMppCurseWPF/Models/Destinatie/ValidatorDestinatie.cs
using CurseMPP.Models;

namespace LabMppCurseWPF.Models
{	
	/// <summary>
	/// Validator pentru Destinatie
	/// </summary>
	public class ValidatorDestinatie : IValidator<Destinatie>
	{	
		/// <summary>
		/// Functie care valideaza o destinatie, arunca ValidationException daca entitatea nu e valida
		/// </summary>
		/// <param name="entity">Destinatia de validat</param>
		public void Validate(Destinatie entity)
		{	
			var errorMessage = "";

			if (entity.Nume.CompareTo("") == 0)
			{
				errorMessage += "Numele unei destinatii nu poate fi vid! ";
			}

			if (errorMessage.CompareTo("") != 0)
			{
				throw new ValidationException(errorMessage);
			}
		}
	}
}

File: mpp/LabMPPCurse/Models/src/main/java/Models/Destinatie/ValidatorDestinatie.java
package Models.Destinatie;

import Models.ValidationException;
import Models.Validator;

public class ValidatorDestinatie implements Validator<Destinatie>{


    @Override
    public void validate(Destinatie entity) throws ValidationException {

        String msg = "";
        if (entity.getNume().compareTo("") == 0)
        {
            msg += "numele e vid";
        }

        if (msg.compareTo("") != 0)
        {
            throw new ValidationException(msg);
        }
    }
}
